tools/spades.py

```diff
@@ -0,0 +1,82 @@
+'''
+    SPAdes - St. Petersburg Assembler
+
+'''
+
+import logging
+import os
+import os.path
+import subprocess
+import shutil
+import random
+import shlex
+import tempfile
+
+import tools
+import tools.samtools
+import tools.picard
+import util.file
+import util.misc
+
+TOOL_NAME = 'spades'
+TOOL_VERSION = '3.10.1'
+
+log = logging.getLogger(__name__)
+
+class SpadesTool(tools.Tool):
+    '''Tool wrapper for SPAdes tool (St. Petersburg Assembler)'''
+
+    def __init__(self, install_methods=None):
+        if install_methods is None:
+            install_methods = [tools.CondaPackage(TOOL_NAME, version=TOOL_VERSION, executable='spades.py',
+                                                  verifycmd='echo spades.py --test', env='spades_env')]
```

Comment:
Is spades given its own conda environment because of weird python dependencies?

Reply:
I recall seeing problems that were fixed by using a separate env, but not what the problems exactly were.  spades is partly in C++ so it may be about C++ library dependencies.  I can look again into moving it into the common env.

Reply:
Don't worry about it for this PR, let's get it merged in and deal with it later if it ends up being possible.